handbook/engineering/search/index.md

```diff
@@ -2,9 +2,93 @@
 
 We own the end-to-end Sourcegraph search experience.
 
+## Vision
+
+Sourcegraph search is:
+
+- **Fast:** Performance of showing results is fast and I can get to what I’m looking for quickly.
+- **Expressive:** It is possible to construct a search query to find exactly the results I’m looking for. The tools, syntax, filters, are provided to construct the queries I want to write.
+The answer to “Can I do ‘X’ with Sourcegraph?” will always be yes.
+- **Easy to use:** Users can figure out how to find what they are looking for, the syntax is clear, new users understand what options they have for searching.
+- **Sharable:** Sourcegraph search creates network effects that provide compounding value with scale. 
```

Comment:
I agree with the value of making search shareable. It's not clear how this fits into the vision. "network effects that provide compounding value with scale." -> sounds very buzz wordy biz speak. I would venture the average user developer doesn't care ("so what?") and for Sourcegraph engineers working on search, I'm not sure this attribute guides the work (does it mean we should create links to share?). I think this could be a tag line for investors, but I'm not sure that's the target audience of this page. Maybe it's intended to mean something else, in which case, a rephrase?

Reply:
Addendum: I see now shareable refers to the code monitoring/saved searches work at the bottom :-) I feel like the code monitoring/search API use is much more aligned with "code insights" than core search concerns (code insights also very much benefits from search functionality, and we have requests to support certain filters, for example). The "improve diff search performance" aspect of code monitoring is core to the "make search fast", but I'm not sure about the rest.

Reply:
I can see how this phrasing for “Shareable” isn’t clear. Shareable search here refers to growing adoption of search through sharing searches that provide team and organization level value. 
Suggested rephrasing: 
> Sourcegraph searches can be easily shared, and provide team and organization wide value. This in turn creates network effects that help compound the value Sourcegraph provides with scale. 

How does this sound? Is it clear how this fits into the vision?

I definitely expect code monitoring to evolve to be a cross-functional effort (somewhat similar to code-intel aware search in the longer term). The framework Quinn Keast lays down for code monitoring in [this RFC](https://docs.google.com/document/d/1_R5DgpUkxyZilsJ9vBQm5cvRPT2udc3tZIPg2q3cnZU/edit#) does a great job explaining how this can be expanded to include campaigns and code insights in the future. 
The first leg of this work is targeted towards search queries. Enabling code monitoring for search queries helps users/teams/orgs keep track of important code changes in their codebases. Notifying users in channels outside of Sourcegraph raises awareness and adoption for these search queries + search results. Like you pointed out, performance considerations like diff search will be an important part of solving this use case for search queries.